lib/fileAction.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * @copyright Copyright (c) 2021 John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @author John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { Node } from "./files/node"
+import logger from "./utils/logger"
+
+interface FileActionData {
+	/** Unique ID */
+	id: string
+	/** Translatable string displayed in the menu */
+	displayName: (files: Node[]) => string
+	/** Svg as inline string. <svg><path fill="..." /></svg> */
+	iconSvgInline: (files: Node[]) => string
+	/** Condition wether this action is shown or not */
+	enabled?: (files: Node[], view) => boolean
+	/**
+	 * Function executed on single file action
+	 * @returns true if the action was executed, false otherwise
+	 * @throws Error if the action failed
+	 */
+	exec: (file: Node) => boolean,
```

Comment:
would it make sense to make these async and awaitable?

Reply:
Done
Also edited first post